src/components/NodeStats.tsx

```diff
@@ -0,0 +1,199 @@
+
+const data = [
+  {
+    x: startDate,
+    y: 1,
+  },
+  {
+    x: startDate + 1 * ONE_DAY,
+    y: 5,
+  },
+  {
+    x: startDate + 2 * ONE_DAY,
+    y: 4,
+  },
+  {
+    x: startDate + 3 * ONE_DAY,
+    y: 12,
+  },
+]
+
+type NodeGraphProps = {
+  streamId?: string,
+}
+
+const NodeGraph = ({ streamId }: NodeGraphProps) => {
+  // TODO: subscribe to stream, display data
+
+  /*
+  useSubscription({
+    stream: streamId,
+    resend: {
+      last: 1,
+    },
+  }, onMessage)
+  */
+
+  return (
+    <Graphs.TimeSeries
+      graphData={{ data }}
+      height="200px"
+      ratio="1:2"
+      showCrosshair
+    />
+  )
+}
+
+type Props = {
+  id?: string,
+}
+
+const NodeGraphLoader = ({ id }: Props) => {
+  const [hasLoaded, setHasLoaded] = useState(false)
+  const [error, setError] = useState<string | undefined>(undefined)
+  const [interval, setInterval] = useState<string>('24hours')
+  const isMounted = useIsMounted()
+
+  const metricStreamId = useMemo(() => `${id}/streamr/node/metrics/${interval}`, [id, interval])
+  // const metricStreamId = 'Y1gWr4X9S8mQdg5mzBq1dA'
+
+  const loadStream = useCallback(async (nodeId) => {
+    setHasLoaded(false)
+    setError(undefined)
+
+    try {
+      await getStream({ id: nodeId })
+    } catch (e) {
+      setError('Metric data not available')
+    } finally {
+      if (isMounted()) {
+        setHasLoaded(true)
+      }
+    }
+  }, [isMounted])
+
+  useEffect(() => {
+    loadStream(metricStreamId)
+  }, [loadStream, metricStreamId])
+
+  return (
+    <>
+      <Graphs defaultInterval="24hours">
+        {(!hasLoaded || !!error) && (
+          <>
+            <GraphPlaceholder>
+              {!!error && (
+                <span>{error}</span>
+              )}
+            </GraphPlaceholder>
+            <Graphs.Loading loading={!error} row={2} />
+          </>
+        )}
+        {!!hasLoaded && !error && (
+          <NodeGraph streamId={metricStreamId} />
+        )}
+        <Graphs.Intervals
+          options={['24hours', '1month', '3months', 'all']}
+          onChange={setInterval}
+        />
+      </Graphs>
+      {!!error && (
+        <Error>
+          {error}
+        </Error>
+      )}
+    </>
+  )
+}
+
+const NodeStats = ({ id }: Props) => {
+  const [selectedStat, setSelectedStat] = useState<string | undefined>(undefined)
+
+  const toggleStat = useCallback((name) => {
+    setSelectedStat((prev) => prev !== name ? name : undefined)
+  }, [])
+
+  const [messagesPerSecond, setMessagesPersecond] = useState<number | undefined>(undefined)
+  const isMounted = useIsMounted()
+
+  const onMessagesPerSecond = useCallback(({
+    eventsPerSecond,
+  }) => {
+    if (isMounted()) {
+      setMessagesPersecond(eventsPerSecond)
+    }
+  }, [isMounted])
+
+  // TODO: use network metric for now, replace with real value
+  // const metricStreamId = useMemo(() => `${id}/streamr/node/metrics/sec`, [id])
+  const metricStreamId = 'Y1gWr4X9S8mQdg5mzBq1dA'
```

Comment:
Comment this out to see how it behaves, it will start spitting out errors in console... it should be smarter also in the stats values if stream is not found 🤔